apps/wallet/src/ui/app/components/ledger/ConnectLedgerModal.tsx

```diff
@@ -58,14 +94,26 @@ export function ConnectLedgerModal({
                     <Button
                         variant="outline"
                         text="Continue"
-                        onClick={onConfirm}
+                        onClick={onContinueClick}
+                        loading={isConnectingToLedger}
                     />
                 </div>
             }
         />
     );
 }
 
+function getErrorMessageForFailedConnection(error: unknown) {
```

Comment:
IMO this seems more like a detail of the consumer (e.g. you'd just onError with the raw error and then this can be done the consuming component), but not a big deal at all.